File: ch6/51.js
/* nCr */

const rl = require('readline-sync')

let n = questionInt("n : "))
let r = questionInt("r : "))

// recursively
let fact = function(x) {
    if (x==0 || x==1) return 1
    else return x * fact(x-1)
}

console.log(fact(n)/(fact(r)*fact(n-r)))

File: ch6/51.cpp
#include <iostream>
using namespace std;

long double fact(long double  x) {
    if(x == 1 || x == 0) return 1;
    else return x * fact(x-1);
}

int main() {
    int n, r; cin >> n >> r;
    cout << fact(n) <<'\n'<< fact(r) <<'\n';
    cout << fact(n) / (fact(r) * fact(n-r));
}
